### DISCUSSION TITLE
Encryption At Rest for privateGPT

### DISCUSSION THREAD

**Discussion by johnbrisbin** [imartinez/privateGPT#481] [Ideas]
If it is not encrypted, how private can it be?
Unfortunately, it won't be easy.
Just so you know,
privateGPT uses the Chroma vector database to manage embeddings.
Chroma uses DuckDB to handle the storage internally.
DuckDB makes things persistent on disk using the Parquet Format.
The Parquet format does support encryption. This is the best part of the story because,
DuckDB says they have no plans to adopt the encryption supported in parquet.
Chroma has nothing in their roadmap about encryption.
And this is the only comment about encryption here.

In short, you probably won't see it very soon.

**Comment by kai11** [comment]
I'm using ecrypted local volumes for documents and data.
LUKS volume is mounted when I start app (before containers) and unmounted after Im done.
Cumbersome, but better than nothing for infrequently used hosted installations.
PS. gocryptfs is a good option too.
